pkg/addon/addon_test.go

```diff
@@ -367,13 +379,53 @@ func TestRenderApp4ObservabilityWithK8sData(t *testing.T) {
 	}
 	for _, tc := range testcases {
 		t.Run("", func(t *testing.T) {
-			app, err := RenderApp(ctx, k8sClient, &tc.addon, nil, tc.args)
+			app, err := RenderApp(ctx, &tc.addon, nil, k8sClient, tc.args)
 			assert.Equal(t, tc.err, err)
 			if app != nil {
 				data, err := json.Marshal(app)
-				assert.NilError(t, err)
+				assert.NoError(t, err)
 				assert.Equal(t, tc.application, string(data))
 			}
 		})
 	}
 }
+
+func TestGetPatternFromItem(t *testing.T) {
+	ossR, err := NewAsyncReader("http://ep.beijing", "some-bucket", "some-sub-path", "", ossType)
+	assert.NoError(t, err)
+	gitR, err := NewAsyncReader("https://github.com/oam-dev/catalog", "", "addons", "", gitType)
```

Comment:
is this request for the real resources in Github?

Reply:
No, this test only use the information in reader to "GetPatternFromItem"

Reply:
good